src/lib/support/BitFlags.h

```diff
@@ -102,7 +102,7 @@ class BitFlags
      */
     BitFlags & Clear(const BitFlags & other)
     {
-        mValue &= ~other.mValue;
+        mValue &= static_cast<IntegerType>(~static_cast<IntegerType>(other.mValue));
```

Comment:
Is the inner cast needed? `mValue` is already `IntegerType/StorageType`.

Reply:
https://github.com/project-chip/connectedhomeip/pull/30680#discussion_r1407820657